node/src/actors/chain_manager/mining.rs

```diff
@@ -168,72 +176,77 @@ impl ChainManager {
             .get_dr_output_pointers_by_epoch(current_epoch);
 
         for dr_pointer in dr_pointers {
-            let data_request_output = self
+            if let Some(data_request_output) = self
                 .chain_state
                 .data_request_pool
-                .get_dr_output(&dr_pointer);
-
-            if data_request_output.is_some() && verify_poe_data_request() {
-                let data_request_output = data_request_output.unwrap();
-                let rad_request = data_request_output.data_request.clone();
-
-                // Send ResolveRA message to RADManager
-                let rad_manager_addr = System::current().registry().get::<RadManager>();
-                rad_manager_addr
-                    .send(ResolveRA {
-                        rad_request,
-                    })
+                .get_dr_output(&dr_pointer)
+            {
+                signature_mngr::vrf_prove(VrfMessage::data_request(beacon, dr_pointer)).map_err(move |e| error!("Couldn't create VRF proof for data request {}: {}", dr_pointer, e))
                     .into_actor(self)
-                    .then(|result, _, _| match result {
-                        Ok(Ok(value)) => fut::ok(value),
-                        Ok(Err(e)) => {
-                            log::error!("Couldn't resolve rad request: {}", e);
-                            fut::err(())
-                        }
-                        Err(e) => {
-                            log::error!("Couldn't resolve rad request: {}", e);
-                            fut::err(())
-                        }
-                    })
-                    .and_then(move |reveal_value, act, _ctx| {
-                        // Create commitment transaction
-                        let commit_body = create_commit_body(dr_pointer, reveal_value.clone());
-                        sign_transaction(&commit_body, 1)
-                            .map_err(|e| log::error!("Couldn't sign commit body: {}", e))
-                            .into_actor(act)
-                            .and_then(move |commit_signatures, act, _ctx| {
-                                let reveal_body = create_reveal_body(dr_pointer,reveal_value, own_pkh);
+                    .and_then(move |proof, act, _ctx| {
+                        let vrf_proof_dr = DataRequestEligibilityClaim {
```

Comment:
Put a comment to specify that DataRequestEligibilityClaim is valid before to create the commit